src/mito/src/engine/procedure/alter.rs

```diff
@@ -394,6 +423,31 @@ mod tests {
         assert!(new_schema.column_schema_by_name("my_field").is_some());
         assert_eq!(new_schema.version(), schema.version() + 1);
         assert_eq!(new_meta.next_column_id, old_meta.next_column_id + 2);
+
+        // Alter the table.
+        let request = new_add_columns_req_with_location();
+        let mut procedure = table_engine
+            .alter_table_procedure(&engine_ctx, request.clone())
+            .unwrap();
+        procedure_test_util::execute_procedure_until_done(&mut procedure).await;
+
+        // Validate.
+        let table = table_engine
+            .get_table(&engine_ctx, &table_ref)
+            .unwrap()
+            .unwrap();
+        let new_info = table.table_info();
+        let new_meta = &new_info.meta;
+        let new_schema = &new_meta.schema;
+
+        assert_eq!(&[0, 1, 6], &new_meta.primary_key_indices[..]);
+        assert_eq!(&[2, 3, 4, 5, 7], &new_meta.value_indices[..]);
+        assert!(new_schema.column_schema_by_name("my_tag_first").is_some());
```

Comment:
Please assert that `my_tag_first` is at the first index and `my_field_after_ts` is after `ts` column.